### PR TITLE
Update the minimal GO version for API module

### PR DESCRIPTION
**Pull request by kkovaletp**
Update GO version in the `go.mod` and read it by the job to set up the testing environment for unit tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -51,14 +51,18 @@ jobs:
...
         id: go
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
+        with:
+          go-version-file: ${{ github.workspace }}/api/go.mod
+          cache: false
 
       - name: Cache Go dependencies
         uses: actions/cache@v2
```
Comment (googollee) [inline .github/workflows/tests.yml:68]: I see you updated this to `v4` in the comment. Do you want to do the same here?

Comment (googollee) [reply .github/workflows/tests.yml:68]: As it's already updating the workflow, I think we could update all dependencies to the latest version.

Comment (kkovaletp) [reply .github/workflows/tests.yml:68]: I thought that it is better to create another PR for that to not mix things together: in addition to updating the versions I planned to migrate at least independent serverless tools from my repo here as well to increase the code testing and lining level

**Review by googollee — APPROVED** [review]
LGTM

**Pull request merged**